### PR TITLE
Target .NET Standard 2.0

### PR DESCRIPTION
**Pull request by natemcmaster**
:watch: blocked on httpsys conversion to ns2.0 cc @Tratcher 

Also updated samples that used obsolete logging API.

### CODE REVIEW

File: test/Microsoft.AspNetCore.StaticFiles.FunctionalTests/Microsoft.AspNetCore.StaticFiles.FunctionalTests.csproj [hunk 1]
```diff
@@ -3,7 +3,15 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFrameworks>netcoreapp2.0;net461</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netcoreapp2.0</TargetFrameworks>
+
+    <!--
+      Workaround for "Use executable flags in Microsoft.NET.Test.Sdk" (https://github.com/Microsoft/vstest/issues/792).
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.StaticFiles.FunctionalTests/Microsoft.AspNetCore.StaticFiles.FunctionalTests.csproj]: Can you file an issue in https://github.com/aspnet/External and self-assign so this is tracked?

File: test/Microsoft.AspNetCore.StaticFiles.FunctionalTests/StaticFileMiddlewareTests.cs [hunk 2]
```diff
@@ -48,14 +48,14 @@ public class StaticFileMiddlewareTests
             var baseAddress = "http://localhost:12345";
             var builder = new WebHostBuilder()
                 .UseKestrel()
-                .UseWebRoot(AppContext.BaseDirectory)
+                .UseWebRoot(AppDomain.CurrentDomain.BaseDirectory)
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.StaticFiles.FunctionalTests/StaticFileMiddlewareTests.cs]: Is `AppContext.BaseDirectory` no longer a thing?

Comment (davidfowl) [inline test/Microsoft.AspNetCore.StaticFiles.FunctionalTests/StaticFileMiddlewareTests.cs]: Its the better API to use

File: test/Microsoft.AspNetCore.StaticFiles.FunctionalTests/Microsoft.AspNetCore.StaticFiles.FunctionalTests.csproj (hunk 1 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.StaticFiles.FunctionalTests/Microsoft.AspNetCore.StaticFiles.FunctionalTests.csproj]: https://github.com/aspnet/External/issues/58

File: test/Microsoft.AspNetCore.StaticFiles.FunctionalTests/StaticFileMiddlewareTests.cs (hunk 2 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.StaticFiles.FunctionalTests/StaticFileMiddlewareTests.cs]: Tomato tomoto. They are exactly the same thing

**Comment by Tratcher** [comment]
Unblocked

**Pull request merged**